src/IntelliHR/Validation/Validators/MaxDate.php

```diff
@@ -2,65 +2,51 @@
 
 namespace IntelliHR\Validation\Validators;
 
-use DateTime;
+use Exception;
 use Illuminate\Contracts\Validation\Validator;
-use InvalidArgumentException;
 
 class MaxDate extends AbstractValidator
 {
     /**
-     * Name of the validator
-     *
      * @var string
      */
     public static $name = 'max_date';
 
     /**
-     * Fallback message
-     *
      * @var string
      */
     public static $message = ':attribute must be before :max_date';
 
-    /**
-     * @param           $attribute
-     * @param           $value
-     * @param array     $parameters
-     * @param Validator $validator
-     *
-     * @return bool
-     * @throws InvalidArgumentException
-     */
     public function validateMaxDate(
         $attribute,
         $value,
         array $parameters,
         Validator $validator
-    ) {
+    ): bool {
         $this->requireParameterCount(1, $parameters, self::$name);
+        $format = $parameters[1] ?? null;
+
+        $maxDate = $this->getDateForParameter($parameters[0], $format);
+
+        try {
+            $date = $this->getDateForParameter($value, $format);
+        } catch (Exception $e) {
+            return true;
```

Comment:
Not sure if I'm reading it correctly, but If the date fails to be created with the specified format, you're passing validation?

Reply:
Oh so it just returns false if it fails?

Reply:
Maybe get `getDateForParameter` to throw the exception for you if it's going to return `false`. Makes it a bit easier to read. Because yay, PHP nuances

Reply:
Yes, it should pass validation if the value is invalid. I think it's the job of a separate date_format validator to test if the date is invalid, not max_date (otherwise you'd end up with two validation errors when only one of them is applicable)